vendor/github.com/openshift/library-go/pkg/operator/deploymentcontroller/new_deployment_controller.go

```diff
@@ -0,0 +1,244 @@
+package deploymentcontroller
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/management"
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsinformersv1 "k8s.io/client-go/informers/apps/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	opv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/apiserver/controller/workload"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+type MyController struct {
+	DeploymentController
+	preconditions []func(context.Context) (bool, error)
+}
+
+func NewMyController(
```

Comment:
I'm not really sure if this meets the goal of writing `new controller with DeploymentController API` - ideally the API of the new controller should match the current deployment controller (so for callers the change is transparent) but I was not sure how that could be done since the signature of workload controller constructor is different.